### PR TITLE
Added xform() and xform_inv() function's changes in 4.0 migration docs

### PR DESCRIPTION
**Pull request by ayanchavand**
Fixes the issue: #6785  
Added xform() and xform_inv() function's changes in 4.0 migration docs

### CODE REVIEW

File: tutorials/migrating/upgrading_to_godot_4.rst [hunk 1]
```diff
@@ -437,6 +437,7 @@ table to find its new name.
 - Shortcut's ``is_valid()`` is now ``has_valid_event()``.
 - TileMap's ``world_to_map()`` is now ``local_to_map()``.
 - TileMap's ``map_to_world()`` is now ``map_to_local()``.
+- Transform2D's ``xform()`` and ``xform_inv()`` are replaced by ``*`` operator for matrix multiplication.
```
Comment (AThousandShips) [inline tutorials/migrating/upgrading_to_godot_4.rst:440]: I'd add some details on that `xform` is `mat * vec` and `xform_inv` is `vec * mat`

Comment (ayanchavand) [reply tutorials/migrating/upgrading_to_godot_4.rst]: edited it

**Review by paddy-exe — APPROVED** [review]

**Review by paddy-exe — CHANGES_REQUESTED** [review]
Needs a Squash

**Review by paddy-exe — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Merged. Thanks and congrats on your first merged contribution!

**Pull request merged**